packages/astro/src/runtime/server/astro-env.ts

```diff
@@ -0,0 +1,22 @@
+import type { GetEnv } from '../../env/types.js';
+import { AstroError, AstroErrorData } from '../../core/errors/index.js';
+export { validateEnvVariable } from '../../env/validators.js';
+
+let _getEnv: GetEnv = (key) => process.env[key];
+
+export function setGetEnv(fn: GetEnv) {
+	_getEnv = fn;
+}
+
+export function getEnv(...args: Parameters<GetEnv>) {
+	return _getEnv(...args);
+}
```

Comment:
Maybe I should comment on the RFC, so take this as a suggestion and not a blocker: you should consider making `getEnv` an async function.

There are some environments where retrieving a secret is an async operation, such as AWS. Their SDK provides a package, and their APIs are callbacks/promises.

Reply:
I'm not familiar with AWS, but would it be possible to retrieve those variables before hand, so we can have a sync version of this? e.g. they can be retrieved before calling `setGetEnv()`

Reply:
I suppose we should think of examples on how users can do that  using an adapter

Reply:
Should I make it async or not?

Reply:
As I mentioned at the beginning, it's not a blocker, so no need to change anything. I will add a comment in the RFC Let's keep it as is for now.

FYI, here's the secret manager SDK: https://www.npmjs.com/package/@aws-sdk/client-secrets-manager